jobserver/pipeline_config.py

```diff
@@ -32,6 +34,34 @@ def get_project(org, repo, branch, get_github_api=_get_github_api):
     raise Exception("Could not find project.yaml")
 
 
+def get_codelists_status(
+    org,
+    repo,
+    branch,
+    get_github_api=_get_github_api,
+    get_opencodelists_api=_get_opencodelists_api,
+):
+    github_api = get_github_api()
+    opencodelists_api = get_opencodelists_api()
+
+    codelists_content = (
+        github_api.get_file(org, repo, branch, filepath="codelists/codelists.txt"),
+        github_api.get_file(org, repo, branch, filepath="codelists/codelists.json"),
+    )
+
+    if all(codelists_content):
+        codelists_check = opencodelists_api.check_codelists(*codelists_content)
+        return codelists_check["status"]
+
+    if github_api.get_branch(org, repo, branch) is None:
+        raise Exception(f"Missing branch: '{branch}'")
+
+    if github_api.get_file(org, repo, branch, filepath="codelists") is not None:
+        raise Exception("Could not find codelists.txt or codelists.json")
+
+    return "ok"
```

Comment:
Is there a known state for getting to this return? If so could you add it in a comment?

Reply:
Yes, will do.  If there's no codelists directory in the repo, there are no codelists to check, so we don't care that it couldn't fetch codelists.txt and codelists.json from inside that directory.